sdk/metric/meter.go

```diff
@@ -444,13 +444,6 @@ func (m *meter) RegisterCallback(f metric.Callback, insts ...metric.Observable)
 	reg := newObserver()
 	var errs multierror
 	for _, inst := range insts {
-		// Unwrap any global.
```

Comment:
Do we need to worry about version skew between our API and SDK (old API with new SDK)?  I don't see anything about that in https://github.com/open-telemetry/opentelemetry-go/blob/main/VERSIONING.md.

Reply:
Good question!

First, I doubt that there are any other alternate metrics SDKs out there--I think we would know. It's probably safe to assume https://github.com/lightstep/otel-launcher-go/tree/main/lightstep/sdk/metric is the only one.

If there were others, they would have noticed that asynchronous instruments were not working through globals--I think this has been the case for a long time, maybe when NewCallback() was introduced?

The otel-launcher-go SDK is used extensively inside our company, but there are not many asynchronous instruments and the ones that are aren't registered through globals. This leads me to suspect that basically no one is using an alternate SDK with globals.

I had to look up the history. I wrote the initial version of the global meter implementation, and I had to visit those old PRs to remember how this unwrap() or Unwrap() stuff started. I realize now, that I was imagining that the global meter concealed this detail. In the original code, there was an `Implementation()` method that returned `any` and the SDKs were expected to call it. 

See https://github.com/open-telemetry/opentelemetry-go/blob/d7b3c3d7e7a7c56e9bfa6aa40d03d75e78e3dd3f/api/metric/sdkhelpers.go#L51

(By the way, no one should go reading that old, old code base!! We're in a much nicer place, today!)

At some point along the way, `Implementation() any` which was part of the SDK instrument interface became `Unwrap() metric.Observer` hidden in an internal package, but the SDK was aware of the detail and downcasts to the same type. I see how this could go either way: we could document the `Unwrap()` interface for alternate SDKs to use with callbacks, or we could fix the global package to eliminate this detail.

Still, your question: I can't see any potential for version-skew problems because`internal/global`, `/metric` and `sdk/metric` are in the same Go module.

Reply:
After sleeping on it, I realize that since our SDK depends on the API of the same version, you can't have that kind of version skew.  Thank you for the history lesson!